pkg/observable/channel/observable.go

```diff
@@ -99,3 +105,56 @@ func (obs *channelObservable[V]) goPublish() {
 	// Unsubscribe all observers as they can no longer receive notifications.
 	obs.observerManager.removeAll()
 }
+
+// Merge takes a slice of observables and merges them into a single observable
+// which publishes notifications from each of the observables supplied.
+// If the closeEarly flag is set to true, the merged observable will close
+// and unsubscribe from all observables when the first of them closes.
+func Merge[V any](
+	ctx context.Context,
+	obs []observable.Observable[V],
+	failFast bool,
+) observable.Observable[V] {
+	// Determine whether the merged observable should close as soon as one of
+	// the observables closes that it is merging of continue notifying observers.
+	fastFailCh := make(chan *struct{})
+	if failFast {
+		failFastAtomic.CompareAndSwap(false, failFast)
+	}
+	mergedObs, mergedObsPubCh := NewObservable[V]()
+
+	// Start a goroutine to wait for the fast fail error channel to close and
+	// then call UnsubscribeAll() on the merged observable.
+	// NB: This only occurs if the closeEarly flag is set to true.
+	go func() {
+		for range fastFailCh {
+			// Do nothing wait for the channel to close
+			continue
+		}
+		mergedObs.UnsubscribeAll()
```

Comment:
Based on its usage in `goPublishNotifications`, I think can be simplified:
```suggestion
		// Wait for the channel to close
		case <- fastFailCh
		mergedObs.UnsubscribeAll()
```